timemachine/fe/rbfe.py

```diff
@@ -202,38 +217,80 @@ def setup_initial_states(
     return initial_states
 
 
-def _optimize_coords_along_states(initial_states: List[InitialState]) -> List[np.ndarray]:
+def setup_optimized_initial_state(
+    st: SingleTopology,
+    lamb: float,
+    host_config: Optional[HostConfig],
+    optimized_initial_states: Sequence[InitialState],
+    temperature: float,
+    seed: int,
+) -> InitialState:
+
+    # Use pre-optimized initial state with the closest value of lambda as a starting point for optimization.
 
+    # NOTE: The current approach for generating optimized conformations in `optimize_coordinates` creates a
+    # discontinuity at lambda=0.5. Ensure that we pick a pre-optimized state on the same side of 0.5 as `lamb`:
+    states_subset = [s for s in optimized_initial_states if s.lamb <= 0.5 == lamb <= 0.5]
```

Comment:
This logic seems closest to preserving the current behavior!

Style opinion: extra parens for easier skimming

```suggestion
    states_subset = [s for s in optimized_initial_states if (s.lamb <= 0.5) == (lamb <= 0.5)]
```

Reply:
Good call, turns out this is actually needed for correctness (see https://github.com/proteneer/timemachine/pull/959/commits/d4cf9b9e0e8eda8ae7a63fb7b9f8a67585ef984a)